src/tabs/programming.ts

```diff
@@ -0,0 +1,108 @@
+import './programming.css'
+import { initializeMonacoEditor, getCode } from './common/monaco-editor'
+
+const codeEditorId = 'code-editor'
+const outputId = 'code-output'
+const errorId = 'code-error'
+const executeButtonSelector = '.code-editor-execute'
+
+let pyodide = null
+let pyodideInitializing
+
+const hideBothAreas = () => {
+  document.getElementById(outputId).style.display = 'none'
+  document.getElementById(errorId).style.display = 'none'
+}
+
+const showErrorArea = () => {
+  document.getElementById(outputId).style.display = 'none'
+  document.getElementById(errorId).style.display = 'block'
+}
+
+const showOutputArea = () => {
+  document.getElementById(outputId).style.display = 'block'
+  document.getElementById(errorId).style.display = 'none'
+}
+
+const printStderr = (text: string) => {
+  if (pyodideInitializing) {
+    return
+  }
+
+  document.getElementById(errorId).innerHTML = text
+  showErrorArea()
+}
+
+const getInput = (): string => {
+  return prompt('Enter input string')
+}
+
+const clearStdout = () => {
+  document.getElementById(outputId).innerHTML = ''
+}
+
+const printStdout = (text: string) => {
+  if (pyodideInitializing) {
+    return
+  }
+
+  console.log(text)
+  showOutputArea()
+  const outputElement = document.getElementById(outputId)
+  text = text.replace(/</g, '&lt;')
+  outputElement.innerHTML = outputElement.innerHTML + text + '\n'
+}
+
+const executeCode = () => {
+  clearStdout()
+
+  const code = getCode()
+
+  try {
+    pyodide.runPython(code)
+  } catch (err) {
+    console.log(err)
```

Comment:
tää vois olla `console.error(err)` niin näkyy punasella sit konsolissa jos jotain menee vikaan

Reply:
Tää taitaa itse asiassa olla to python koodin virhe joka printataan jo siihen boksiin. Tarviikohan tätä edes laittaa konsoliin?